MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -1619,12 +1623,26 @@ private void updateLocation(@NonNull LocationUpdate locationUpdate, boolean from
     }
     CameraPosition currentCameraPosition = mapboxMap.getCameraPosition();
     boolean isGpsNorth = getCameraMode() == CameraMode.TRACKING_GPS_NORTH;
-    locationAnimatorCoordinator.feedNewLocation(
-      getTargetLocationWithIntermediates(locationUpdate.getLocation(), locationUpdate.getIntermediatePoints()),
-      currentCameraPosition,
-      isGpsNorth,
-      fromLastLocation ? Long.valueOf(0L) : locationUpdate.getAnimationDuration());
-    updateAccuracyRadius(locationUpdate.getLocation(), false);
+
+
+    //Performance improvement: Check if new location requires a redraw to be performed, taking the pixel tolerance into account.
+    //By default, the tolerance is 1 pixel, meaning that the new location must impact at least a different pixel on the screen to justify a redraw.
+    boolean updateNeeded = (lastLocation == null);
+    if(updateNeeded == false) {
+      PointF lastPoint = this.mapboxMap.getProjection().toScreenLocation(new LatLng(lastLocation.getLatitude(), lastLocation.getLongitude()));
```

Comment:
Can we save the `lastPoint` as a property to save one calculation on it?

Reply:
good idea, i'll do it if possible.

Reply:
It's not possible as the pointF will be deprecated as soon as the map is scroll / rotated, etc...

Reply:
In that scenario, we definitely need to update, right?

Reply:
How about only calculate it from `lastLocation` in that scenario? But I'm ok with the current solution too.

Reply:
if the current solution if acceptable, let's go with this one.